modules/local/appendmetadata/main.nf

```diff
@@ -0,0 +1,31 @@
+process APPEND_METADATA {
+    tag "append_metadata"
+    label 'process_low'
+
+    input:
+    val clusters_path  // cluster data as a TSV path
+                        // this needs to be "val", because "path"
+                        // won't stage the file correctly for exec
+    val metadata  // metadata to be appended, list of lists
+
+    output:
+    path("clusters_and_metadata.tsv"), emit: clusters
+
+    exec:
+    def merged = []
+    def rows
+
+    clusters_path.withReader { reader ->
+        rows = reader.readLines()*.split('\t')
+    }
+
+    for(int i = 0; i < rows.size(); i++)
+    {
+        merged.add(rows[i] + metadata[i])
+    }
+
+    task.workDir.resolve("clusters_and_metadata.tsv").withWriter { writer ->
+        merged.each { writer.writeLine it.join("\t") }
```

Comment:
I believe if a metadata entry contains a tab `\t`, then columns could become mixed up. Is there any other way to write CSV files that e.g., will properly escape or quote the columns?

Alternatively, we could restrict metadata entries to not contain tabs in the JSON Schema, so at least the pipeline would not run with that type of input data.

Reply:
Yeah, I restricted the schema to not allow tabs here (f5c959db104b7eab151bed8ce9c232a8bb901f23).

Reply:
Thanks. However, you applied this to the `nextflow_schema.json`, which means the restriction only applies to the header names. Could you apply this to the `schema_input.json` instead? And could you maybe add a test case to make sure the pipeline fails if metadata contains tabs?

Another solution would be to properly quote any metdata fields containing tabs.

Reply:
Good catch. Updated in 44cfb49bd2ac65c89af3ce55f5218887104a7445